vss-processor/src/main/kotlin/org/eclipse/kuksa/vssprocessor/spec/VssSpecificationSpecModel.kt

```diff
@@ -0,0 +1,314 @@
+import com.squareup.kotlinpoet.TypeName
+import com.squareup.kotlinpoet.TypeSpec
+import com.squareup.kotlinpoet.asClassName
+import com.squareup.kotlinpoet.asTypeName
+import org.eclipse.kuksa.vsscore.model.VssNode
+import org.eclipse.kuksa.vsscore.model.VssProperty
+import org.eclipse.kuksa.vsscore.model.VssSpecification
+import org.eclipse.kuksa.vsscore.model.className
+import org.eclipse.kuksa.vsscore.model.name
+import org.eclipse.kuksa.vsscore.model.variableName
+import kotlin.reflect.KClass
+import kotlin.reflect.KProperty1
+import kotlin.reflect.full.declaredMemberProperties
+
+// Reflects the specification file as a model and is filled via reflection. That is why the variable names
+// should exactly match the names inside the specification file and be of a string type.
+internal class VssSpecificationSpecModel(
+    override var uuid: String = "",
+    override var vssPath: String = "",
+    override var description: String = "",
+    override var type: String = "",
+    override var comment: String = "",
+    @Suppress("MemberVisibilityCanBePrivate") var datatype: String = "",
+) : VssSpecification, SpecModel {
+    var childSpecifications = mutableListOf<VssSpecificationSpecModel>()
+
+    private lateinit var logger: KSPLogger
+    private lateinit var packageName: String
+
+    private val datatypeProperty: TypeName
+        get() {
+            return when (datatype) {
+                "string" -> String::class.asTypeName()
+                "boolean" -> Boolean::class.asTypeName()
+                // Do not use UInt because it is incompatible with @JvmOverloads annotation
+                "uint8", "uint16", "uint32" -> Int::class.asTypeName()
+                "int8", "int16", "int32" -> Int::class.asTypeName()
+                "float" -> Float::class.asTypeName()
+                "double" -> Double::class.asTypeName()
+                "string[]" -> Array::class.parameterizedBy(String::class)
+                "boolean[]" -> BooleanArray::class.asTypeName()
+                "uint8[]", "uint16[]", "uint32[]", "int8[]", "int16[]", "int32[]" -> IntArray::class.asTypeName()
+                else -> String::class.asTypeName()
+            }
+        }
+
+    /**
+     * Returns valid default values as string literals.
+     */
+    private val defaultValue: String?
+        get() {
+            return when (datatypeProperty) {
+                String::class.asTypeName() -> "\"\""
+                Boolean::class.asTypeName() -> "false"
+                Float::class.asTypeName() -> "0f"
+                Double::class.asTypeName() -> "0.0"
+                Int::class.asTypeName() -> "0"
+                Long::class.asTypeName() -> "0L"
+                UInt::class.asTypeName() -> "0u"
+                Array::class.parameterizedBy(String::class) -> "emptyArray<String>()"
+                IntArray::class.asTypeName() -> "IntArray(0)"
+                BooleanArray::class.asTypeName() -> "BooleanArray(0)"
+
+                else -> null
```

Comment:
can you test this behavior with different Arrays (e.g. String, Int, Boolean, CustomType)?

Reply:
I will try ;)